source/common/network/connection_impl.cc

```diff
@@ -49,7 +49,8 @@ ConnectionImpl::ConnectionImpl(Event::Dispatcher& dispatcher, ConnectionSocketPt
       socket_(std::move(socket)), write_buffer_(dispatcher.getWatermarkFactory().create(
                                       [this]() -> void { this->onLowWatermark(); },
                                       [this]() -> void { this->onHighWatermark(); })),
-      dispatcher_(dispatcher), id_(next_global_id_++) {
+      dispatcher_(dispatcher), id_(next_global_id_++),
+      per_connection_state_(Envoy::RequestInfo::FilterStateImpl()) {
```

Comment:
This stores a reference to a temporary. I don't think you really want a reference here.

Reply:
Oops! I hacked up something quickly. Will clean all this up.